docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,350 @@
+# Gluster-Block Provisioning End-To-End Design
+
+## Motivation
+
+While GlusterFS volumes can be used for RWO as well as RWX (and ROX) volumes,
+for some workloads, specifically metadata-heavy, the RWO support is not
+satisfactory with respect to performance. Furthermore, the scalability
+in number of volumes is currently not very good.
+
```

Comment:
I think the scalability issue is well framed but I think the rest of the problem description/motivation is a little misleading. I think its important to clarify to folks inside and outside the community so they can understand what problem we're really solving here. In reality today, the gluster deployment that comes with gluster-kubernetes only supports ReadWriteMany volumes(RWX). RWO volumes are defined as only supporting a single concurrent writer and being locked to a host. While it is possible to use any RWX volume in place of a RWO volume, it does not provide the single writer or locking semantics that one expects when using a RWO volume. It is a bad idea to use a RWX volume in place of a RWO volume, because the Kubernetes user is unaware and they think they are getting the RWO semantics to protect their data from corruption and really they are not. Thus, this design PR is intended to add real RWO support to gluster-kubernetes.